src/cargo/util/toml/targets.rs

```diff
@@ -713,14 +726,20 @@ fn target_path(
     match (first, second) {
         (Some(path), None) => Ok(path),
         (None, None) | (Some(_), Some(_)) => {
+            let err = || {
+                Err(format!(
+                    "can't find `{name}` {target_kind}, specify {target_kind}.path",
+                    name = name,
+                    target_kind = target_kind
+                ))
+            };
+            if edition >= Edition::Edition2018 {
+                return err();
+            }
             if let Some(path) = legacy_path(target) {
```

Comment:
Hm, let's just wrap this if-let into an addition `if edition >= 2018`, so as to avoid introducing local closures?

Reply:
of course.. I find out you can't use && with `if let` and somehow forgot about just nesting 😆

also I assumed the closure was free. good to know it's not.